src/ol/style/fill.js

```diff
@@ -31,13 +31,14 @@ ol.style.Fill = function(opt_options) {
 
 
 /**
- * Clones the style.
+ * Clones the style. The color is not cloned if it is an {@link ol.ColorLike}.
  * @return {ol.style.Fill} The cloned style.
  * @api
  */
 ol.style.Fill.prototype.clone = function() {
+  var color = this.getColor();
   return new ol.style.Fill({
-    color: (this.getColor() instanceof Array) ? this.getColor().slice(0) : this.getColor()
+    color: (color && color.slice) ? color.slice() : color || undefined
```

Comment:
I backed down from using an ol.color.clone method because sometimes the type to clone is {ol.Color|string} and sometimes it {ol.Color|ol.ColorLike} and it was easier this way. I could use ol.colorlike.isColorLike, though.